querybook/webapp/lib/datasource.ts

```diff
@@ -156,10 +156,44 @@ export function uploadDatasource<T = null>(
     return syncDatasource<T>('POST', urlOptions, null, options);
 }
 
+async function streamDatasource(
+    url: string,
+    data?: Record<string, unknown>,
+    onData?: (data: string) => void,
```

Comment:
there is data and onData which is a bit confusing? 
also would be nice to document that the data passed in for onData is the total of current data instead of the partial delta